packages/grid/src/vaadin-grid-data-provider-mixin.js

```diff
@@ -25,8 +25,8 @@ export const ItemCache = class ItemCache {
     this.parentItem = parentItem;
     /** @type {object} */
     this.itemCaches = {};
-    /** @type {object} */
-    this.items = {};
+    /** @type {object[]} */
+    this.items = [];
```

Comment:
It turns out that the cache items have been an object since long time ago. There is no reason to use an object instead of a simple array.